sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerDriverFactory.java

```diff
@@ -93,11 +97,44 @@ protected SchedulerDriver createInternal(
             final FrameworkInfo frameworkInfo,
             final String masterUrl,
             final Credential credential) {
-        if (credential == null) {
-            return new MesosToSchedulerDriverAdapter(scheduler, frameworkInfo, masterUrl, true);
-        } else {
-            return new MesosToSchedulerDriverAdapter(scheduler, frameworkInfo, masterUrl, true, credential);
+        Capabilities capabilities = Capabilities.getInstance();
+        if (credential != null) {
+            return new MesosToSchedulerDriverAdapter(scheduler, frameworkInfo, masterUrl, true, credential) {
+                @Override
```

Comment:
Should we not subclass `MesosToSchedulerDriverAdapter` instead of doing it inline like this?

Reply:
A lambda won't work here since you can't wire up a lambda to have access to object scope, so no reference to `this` is possible (this is because lambdas are actually just syntactic sugar for implementors of an interface with just one unimplemented method, viz [FunctionalInterface](https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html)).

I made it inline kind of purposefully -- this really shouldn't be something that sticks around, and reifying it into an actual class instead of an anonymous object makes that intent clearer IMO.